mitodl/mitxonline
# feat: Change certificate generation based on certificate available date

#### Pre-Flight checklist


#### What are the relevant tickets?
Part of https://github.com/mitodl/mitxonline/issues/1257

#### What's this PR do? (As of now, this PR enables us to at least add the certificate end dates manually until we sync them with edX)
- Adds a new field `certificate_available_date`
- Bases the certificate generation functionality on `certificate_available_date` instead of `end_date`
- Moving on we need to sync this date with edX (See https://github.com/mitodl/mitxonline/issues/1257#issuecomment-1346463078)
 

#### How should this be manually tested?
You will need a few configurations and setting up things to test this PR end-to-end.

- Try generating the certificate with steps mentioned in https://github.com/mitodl/mitxonline/pull/722 with `certificate_available_date` instead of `end_date`.
- The behavior that was once based on the course run's `end_date` should now be transferred to `certificate_available_date`. The existing behavior is mentioned in detail at https://github.com/mitodl/mitxonline/pull/722

#### Where should the reviewer start?
Configure your edX instance with MITxOnline Locally

#### Any background context you want to provide?
We added the polling and certificate generation in https://github.com/mitodl/mitxonline/pull/722 where we based the conditions on course end_date

**NOTE**
Probably, there are going to be some other PRs as well that will enable us to sync the `certificate_available_date` with edX. But we are not yet ready to start on them, See the comments (https://github.com/mitodl/mitxonline/issues/1257#issuecomment-1354905830, https://github.com/mitodl/mitxonline/issues/1257#issuecomment-1355023307)

#### Screenshots (if appropriate)
<img width="799" alt="image" src="https://user-images.githubusercontent.com/34372316/209666996-9d084af1-6f50-49ba-959a-90c0dda8257e.png">
